### PR TITLE
Hide next anchorID

### PR DESCRIPTION
**Pull request by vimukthi-git**
#centrifuge/go-centrifuge/778

### CODE REVIEW

File: contracts/AnchorRepository.sol [hunk 1]
```diff
@@ -66,41 +66,44 @@ contract AnchorRepository is Initializable {
...
   /**
-   * @param _anchorId Id for an Anchor.
+   * @param _unHashedAnchorId secret anchorID for an Anchor.
    * @param _documentRoot merkle tree for a document that will be anchored/commited. It also contains the signatures
    * @param _documentProofs Array containing proofs for the document's signatures.
    * The documentRoot must be a merkle tree constructed from the signingRoot plus all signatures
    */
   function commit(
-    uint256 _anchorId,
+    uint256 _unHashedAnchorId,
```
Comment (rdinicut) [inline contracts/AnchorRepository.sol]: I would call this anchorSecret and the hashed version just anchorId

File: contracts/AnchorRepository.sol [hunk 2]
```diff
@@ -66,41 +66,44 @@ contract AnchorRepository is Initializable {
...
     bytes32 _documentRoot,
     bytes32[] calldata _documentProofs
   )
   external
   payable
   {
+
+    uint256 hashedAnchor = uint256(sha256(abi.encodePacked(_unHashedAnchorId)));
```
Comment (rdinicut) [inline contracts/AnchorRepository.sol]: we want to use sha256 not keccak256?

Comment (vimukthi-git) [reply contracts/AnchorRepository.sol]: it was preferred by @lucasvo

Comment (lucasvo) [reply contracts/AnchorRepository.sol]: We use sha256 everywhere else. Our tree is constructed with it. Makes zk proofs much simpler to do. I'd replace 'unhashed' with anchorIdPreImage
```suggestion
    uint256 hashedAnchor = uint256(sha256(abi.encodePacked(_anchorIdPreImage)));
```

Comment (lucasvo) [reply contracts/AnchorRepository.sol]: I actually renamed this in the issue already I think.

Comment (manuelpolzhofer) [reply contracts/AnchorRepository.sol]: something about the naming convention.
For me I see the naming in the following way:
anchor = documentRoot
anchorId = key in the map to get the documentRoot

I would rename `hashedAnchor` into `anchorId`.

**Pull request merged**